src/state/slices/foxEthSlice/selectors.ts

```diff
@@ -0,0 +1,73 @@
+import { createSelector } from '@reduxjs/toolkit'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { toBaseUnit } from 'lib/math'
+import type { ReduxState } from 'state/reducer'
+import { createDeepEqualOutputSelector } from 'state/selector-utils'
+import { selectAssets } from 'state/slices/assetsSlice/selectors'
+import { selectMarketData } from 'state/slices/marketDataSlice/selectors'
+
+import { foxEthLpAssetId } from './constants'
+
+export const selectFoxEthLpOpportunity = (state: ReduxState) => state.foxEth.lpOpportunity
+
+export const selectFoxFarmingOpportunities = (state: ReduxState) =>
+  state.foxEth.farmingOpportunities
+
+export const selectVisibleFoxFarmingOpportunities = createDeepEqualOutputSelector(
+  selectFoxFarmingOpportunities,
+  opportunities =>
+    opportunities.filter(
+      opportunity =>
+        !opportunity.expired || (opportunity.expired && bnOrZero(opportunity.cryptoAmount).gt(0)),
+    ),
+)
+
+export const selectFoxFarmingOpportunityByContractAddress = createSelector(
+  selectFoxFarmingOpportunities,
+  (_state: ReduxState, contractAddress: string) => contractAddress,
+  (opportunities, contractAddress) =>
+    opportunities.find(opportunity => opportunity.contractAddress === contractAddress),
+)
+
+export const selectFarmContractsBalance = createDeepEqualOutputSelector(
```

Comment:
Could be a regular selector output comparison since it will use triple equal comparison on strings